working-docs/transient.md

```diff
@@ -0,0 +1,379 @@
+| [1037](https://cwe.mitre.org/data/definitions/1037.html) | Processor Optimization Removal or Modification of Security-critical Code          | The developer builds a security-critical protection mechanism into the software, but the processor optimizes the execution of the program such that the mechanism is removed or modified. |
+| [1264](https://cwe.mitre.org/data/definitions/1264.html) | Hardware Logic with Insecure De-Synchronization between Control and Data Channels | The hardware logic for error handling and security checks can incorrectly forward data before the security check is complete.                                                             |
+| [1303](https://cwe.mitre.org/data/definitions/1303.html) | Non-Transparent Sharing of Microarchitectural Resources                           | Hardware structures shared across execution contexts (e.g., caches and branch predictors) can violate the expected architecture isolation between contexts.                               |
+| [1342](https://cwe.mitre.org/data/definitions/1342.html) | Information Exposure through Microarchitectural State after Transient Execution   | The processor does not properly clear microarchitectural state after incorrect microcode assists or speculative execution, resulting in transient execution.                              |
+
+These existing CWEs do not address the root causes of transient
+execution attacks, and therefore do not suffice to characterize them.
+For example, consider
+[CVE-2017-5753](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2017-5753)
+(Bounds Check Bypass, or BCB):
+
+  - Although BCB can affect conditional branches that do serve as
+    “security-critical protection mechanisms” (for example, in
+    software sandboxes), BCB can also affect branches that serve a
+    functional purpose, such as dynamic type checking \[2\]. Therefore
+    CWE-1037 is too narrow.
+
+  - The conditional branch instructions affected by BCB are not
+    “hardware logic for error handling and security checks.” Hence
+    CWE-1264 does not apply.
+
+  - BCB can be exploited over a network \[3\] without shared hardware
+    resources, and therefore CWE-1303 does not apply.
+
+  - CWE-1342 implies that the processor should be clearing
+    microarchitectural state, which is impractical or infeasible for
+    many vulnerabilities, including BCB.
+
+The objective of this document is to define
+**<span class="underline">four</span>** new CWEs that address the root
+causes of transient execution attacks, and with language that is
+sufficiently generic to benefit the entire CWE community.
+
+New CWE Proposals
+--------------------------------
+
+### CWE-A: Processor Event Causes Transient Execution
```

Comment:
Steven: I'm confused about why this is a concern. Isn't transient execution a normal, expected behavior? So, being able to cause it doesn't seem like an issue. Is the key issue about "allowing transient execution with microarchitectural side effects that can be observed by an adversary"?